src/bootstrap/setup.rs

```diff
@@ -1,11 +1,56 @@
 use crate::t;
 use std::path::{Path, PathBuf};
+use std::str::FromStr;
 use std::{
-    env, fs,
+    env, fmt, fs,
     io::{self, Write},
 };
 
-pub fn setup(src_path: &Path, include_name: &str) {
+#[derive(Clone, Copy)]
+pub enum Profile {
+    Compiler,
+    Codegen,
+    Library,
+    User,
+}
+
+impl Profile {
+    fn include_path(&self, src_path: &Path) -> PathBuf {
+        PathBuf::from(format!("{}/src/bootstrap/defaults/config.toml.{}", src_path.display(), self))
+    }
+}
+
+#[derive(Debug)]
+pub struct ProfileErr {
+    pub name: String,
+}
+
+impl FromStr for Profile {
+    type Err = ProfileErr;
```

Comment:
The custom error may be overkill, could also just return the "wrong" profile name as a `String`

Reply:
I think a `String` would be fine, yeah. The additional type isn't adding more info so I don't think it's necessary.